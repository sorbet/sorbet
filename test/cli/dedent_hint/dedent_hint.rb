# typed: false

class A
  def test0
    x = nil
    if
  end

  def test1
    x = nil
    if x
  end

  def test2
    x = nil
    if x.
  end

  def test3
    x = nil
    if x.f
  end

  # -- These should still have no errors even in indentationAware mode --

  def no_syntax_error_1
    x = if y
    end
  end

  def no_syntax_error_2
    x = if y
        end
  end

  def no_syntax_error_3
      # this is a comment with weird indent
    x = if y
    end
    puts 'after'
  end
end
