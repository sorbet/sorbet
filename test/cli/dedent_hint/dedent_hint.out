test/cli/dedent_hint/dedent_hint.rb:6: unexpected token "if" https://srb.help/2001
     6 |    if
            ^^

test/cli/dedent_hint/dedent_hint.rb:17: unexpected token "end" https://srb.help/2001
    17 |  end
          ^^^

test/cli/dedent_hint/dedent_hint.rb:42: unexpected token "end of file" https://srb.help/2001
    42 |end
    43 |

test/cli/dedent_hint/dedent_hint.rb:11: Hint: this "if" token might not be properly closed https://srb.help/2003
    11 |    if x
            ^^
    test/cli/dedent_hint/dedent_hint.rb:12: Matching `end` found here but is not indented as far
    12 |  end
          ^^^

test/cli/dedent_hint/dedent_hint.rb:16: Hint: this "if" token might not be properly closed https://srb.help/2003
    16 |    if x.
            ^^
    test/cli/dedent_hint/dedent_hint.rb:17: Matching `end` found here but is not indented as far
    17 |  end
          ^^^

test/cli/dedent_hint/dedent_hint.rb:21: Hint: this "if" token might not be properly closed https://srb.help/2003
    21 |    if x.f
            ^^
    test/cli/dedent_hint/dedent_hint.rb:22: Matching `end` found here but is not indented as far
    22 |  end
          ^^^
Errors: 6
